### DISCUSSION TITLE
Scheduled tasks: Am I doing it wrong?

### DISCUSSION THREAD

**Discussion by opicodes** [laravel-zero/laravel-zero#392] [Q&A]
Hi,
I created a command and want to schedule it. I added the required method 

```
    public function schedule(Schedule $schedule): void
    {
        $schedule->command(static::class)->everyMinute();
    }
```

to the command itself.

I then installed the schedule-list component, but running the `schedule:list` command always returns `No tasks scheduled`.

I suppose there is some magic that somehow check which commands exist and schedule them when they have that method?

**Comment by owenvoke** [comment]
I think this is maybe a bug, as they should show up in the `schedule:list` command. As far as I know that's because of [this code](https://github.com/laravel-zero/framework/blob/cc07fa599c9c49d1b8ec280b7d5498edc60eabed/src/Kernel.php#L222) where it runs the `schedule()` command when Artisan boots. Rather than adding them to the list of events which would then be shown in [this code](https://github.com/laravel/framework/blob/b9203fca96960ef9cd8860cb4ec99d1279353a8d/src/Illuminate/Console/Scheduling/ScheduleListCommand.php#L36).

Just to confirm, your command is extending `LaravelZero\Framework\Commands\Command` rather than `Illuminate\Console\Command`? 👍🏻

**Reply by opicodes** [reply]
Thanks very much, you are a life saver, I didn't notice the different parent. I guess it should made more prominent in the docs either in the command section or in the Schedule Tasks one (being a requirement for this).

**Reply by owenvoke** [reply]
No problem. 👍🏻 Yeah, I'll update the docs later. Or feel free to if you'd like. 👍🏻

**Reply by opicodes** [reply]
Another thing that seems to be missing is the schedule command `schedule:work` to mimic the cron for development as described [here](https://laravel.com/docs/9.x/scheduling#running-the-scheduler-locally)

**Comment by surgiie** [comment]
Im having the same issue, scheduler doesnt appear to work. Someone has suggested a PR fix: https://github.com/laravel-zero/framework/pull/462

But it doesnt seem like this project has much maintainers or is delaying merging PRs as there are some dated from January still open. :/
